src/app/pages/AccountDetailsPage/ContractCodeCard.tsx

```diff
@@ -0,0 +1,93 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import Card from '@mui/material/Card'
+import CardContent from '@mui/material/CardContent'
+import { ScrollingDiv } from '../../components/PageLayout/ScrollingDiv'
+import { useAccount } from './hook'
+import { useRequiredScopeParam } from '../../hooks/useScopeParam'
+import { useLoaderData } from 'react-router-dom'
+import { CardEmptyState } from './CardEmptyState'
+import Typography from '@mui/material/Typography'
+import { ScrollingDataDisplay } from '../../components/ScrollingDataDisplay'
+import Box from '@mui/material/Box'
+import { CopyToClipboard } from '../../components/CopyToClipboard'
+import { useScreenSize } from '../../hooks/useScreensize'
+
+export const contractCodeContainerId = 'code'
+
+export const ContractCodeCard: FC = () => {
+  const { t } = useTranslation()
+  const scope = useRequiredScopeParam()
+  const address = useLoaderData() as string
+  const { isMobile } = useScreenSize()
+
+  const { isFetched, account } = useAccount(scope, address)
+  const contract = account?.evm_contract
+  const hasCreation = !!contract?.creation_bytecode
+  const hasRuntime = !!contract?.runtime_bytecode
+  const noCode = isFetched && !hasCreation && !hasRuntime
+  return (
+    <Card>
+      <ScrollingDiv id={contractCodeContainerId}>
+        {noCode && <CardEmptyState label={t('contract.noCode')} />}
+        {(hasCreation || hasRuntime) && (
+          <CardContent>
+            {hasCreation && (
+              <>
+                <Box
+                  sx={{
+                    display: 'flex',
+                    justifyContent: 'space-between',
+                    alignItems: 'center',
+                    mb: '10px',
```

Comment:
In general we rely on theme spacing where possible, because mocks are using totally random sizes. for paddings and margins we try to use the nearest possible theme spacing

mb: 3 // will generate '8px'

Reply:
I see. OK.